3529/1/gemhandler.rb

```diff
@@ -0,0 +1,108 @@
+require 'nokogiri'
+require 'json'
+require 'net/http'
+require 'uri'
+require 'open-uri'
+require 'httparty'
+require 'open_uri_redirections'
+require 'yaml'
+require 'octokit'
+require 'typhoeus'
+
+class GemHendler
+  attr_reader :data_about_gem
+
+  def initialize(github_url, gem_name)
+    @url = github_url
+    @data_about_gem = {}
+    puts "Didn't find repository on github" unless @url
+    @repo_addr = adress_handle
+    @client = Octokit::Client.new(login: 'gannagoodkevich', password: 'Pusivill1999')
+    @client.auto_paginate = true
+    login
+    @data_about_gem[:name] = gem_name
+    join_all_data
+  end
+
+  def login
+    user = @client.user
+    user.login
+  end
+
+  def adress_handle
+    @repo_addr = if @url.include?('https://github.com/')
+                   @url.gsub('https://github.com/', '')
+                 else
+                   @url.gsub('http://github.com/', '')
+                 end
+  end
+
+  def join_all_data
+    find_wsf
+    find_uci
+    @data_about_gem[:rate] = make_rate
+  end
+
+  def find_wsf
+    find_watchers
+    find_stars
+    find_forks
+  end
+
+  def find_uci
+    find_used_by
+    find_contributers
+    find_issues
+  end
+
+  def make_rate
+    rate = find_watch_plus_starts + find_forks_plus_contributors
+    rate + @data_about_gem[:issues] * 0.05 + @data_about_gem[:used_by] * 0.5
+  end
+
+  def find_watch_plus_starts
+    @data_about_gem[:watched_by] * 0.15 + @data_about_gem[:stars] * 0.15
+  end
+
+  def find_forks_plus_contributors
+    @data_about_gem[:watched_by] * 0.15 + @data_about_gem[:stars] * 0.15
+  end
+
+  def find_forks
+    repo = @client.repo @repo_addr
```

Comment:
Although in Ruby you can use methods without brackets, the community prefers to use them